### PR TITLE
Adding a Cyclone trainer tutorial

### PR DESCRIPTION
**Pull request by calebrob6**
This PR adds a new tutorial showing how to use `torchgeo.trainers` with PyTorch Lightning to train and evaluate a simple model on the Cyclone dataset.

### CODE REVIEW

**Comment by calebrob6** [comment]
Going to push another version of the notebook trained for 10 epochs so that readers can see that it achieves a good loss.

File: docs/index.rst [hunk 1]
```diff
@@ -27,6 +27,7 @@ torchgeo
 
    tutorials/getting_started
    tutorials/benchmarking
+   tutorials/trainer_tutorial
```
Comment (adamjstewart) [inline docs/index.rst:30]: This filename is kind of redundant, let's just call it `tutorials/trainers.ipynb`

**Review by adamjstewart — COMMENTED** [review]
Initial comment before I dig into the tutorial itself. 

I'm not really sure the best way to review tutorial PRs. I guess I should just make the changes myself and push them to this branch? I wish it were easier to visualize and comment on notebooks directly on GitHub.

**Comment by calebrob6** [comment]
sounds good to me!

File: docs/index.rst (hunk 1 above)
Comment (calebrob6) [reply docs/index.rst:30]: renaming now

**Comment by adamjstewart** [comment]
This looks great! I made a bunch of very minor grammar fixes ("setup" is an adjective, "set up" is a verb, etc.)

The only big change I made was renaming the tutorial from "Training models" to "PyTorch Lightning trainers". I think "Training models" is a bit too generic since you can train a model with or without PyTorch Lightning. Thoughts on this?

Other musings:
* `data_dir` and `experiment_dir` will be in two separate temp dirs, but they don't necessarily have to be
* the CSV parsing and plotting code seems a bit cumbersome, could we add a plot method to the trainers that automatically plots this kind of thing?
* the output of testing isn't very nicely formatted

**Comment by calebrob6** [comment]
Cool, merge away!

> The only big change I made was renaming the tutorial from "Training models" to "PyTorch Lightning trainers". I think "Training models" is a bit too generic since you can train a model with or without PyTorch Lightning. Thoughts on this?

Fine with me, although I don't think we have immediate plans to write anything other than PyTorch Lightning trainers.

> the CSV parsing and plotting code seems a bit cumbersome, could we add a plot method to the trainers that automatically plots this kind of thing?

These are usually plotted by TensorBoard (this is what happens in the `train.py` script) on the fly, so it isn't a big issue, e.g.:
image

**Comment by adamjstewart** [comment]
> Fine with me, although I don't think we have immediate plans to write anything other than PyTorch Lightning trainers.

I meant just writing your own for-loop over the data and computing the loss yourself. I think that's how most people did things before Lightning, prob still do.

**Review by adamjstewart — APPROVED** [review]
